backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java

```diff
@@ -567,7 +569,16 @@ public ExperimentRun getExperimentRun(String experimentRunId)
         throw StatusProto.toStatusRuntimeException(status);
       }
       LOGGER.debug("Got ExperimentRun successfully");
-      return experimentRunEntity.getProtoObject();
+      ExperimentRun experimentRun = experimentRunEntity.getProtoObject();
+      if (experimentRun.getDatasetsCount() > 0) {
+        experimentRun = checkDatasetVersionBasedOnPrivileges(experimentRun);
+      }
+      if (experimentRun.getVersionedInputs() != null
+          && experimentRun.getVersionedInputs().getRepositoryId() != 0) {
+        experimentRun =
+            checkVersionInputBasedOnPrivileges(experimentRun, new HashSet<>(), new HashSet<>());
+      }
```

Comment:
move this to a function, rest things look fine

Reply:
Moved to separate function